torchgeo/datasets/quakeset.py

```diff
@@ -0,0 +1,291 @@
+    url = "https://hf.co/datasets/DarthReca/quakeset/resolve/bead1d25fb9979dbf703f9ede3e8b349f73b29f7/earthquakes.h5"
+    md5 = "76fc7c76b7ca56f4844d852e175e1560"
+    splits = {"train": "train", "val": "validation", "test": "test"}
+    classes = ["unaffected_area", "earthquake_affected_area"]
+
+    def __init__(
+        self,
+        root: str = "data",
+        split: str = "train",
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new QuakeSet dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of "train", "val", or "test"
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            AssertionError: If ``split`` argument is invalid.
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+        """
+        assert split in self.splits
+
+        self.root = root
+        self.split = split
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+        self.filepath = os.path.join(root, self.filename)
+
+        self._verify()
+
+        try:
+            import h5py  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "h5py is not installed and is required to use this dataset"
+            )
+
+        self.data = self._load_data()
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            sample containing image and mask
+        """
+        image = self._load_image(index)
+        label = torch.tensor(self.data[index]["label"])
+        magnitude = torch.tensor(self.data[index]["magnitude"])
+
+        sample = {"image": image, "label": label, "magnitude": magnitude}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.data)
+
+    def _load_data(self) -> list[dict[str, Any]]:
+        """Return the metadata for a given split.
+
+        Returns:
+            the sample keys, patches, images, labels, and magnitudes
+        """
+        import h5py
+
+        f = h5py.File(self.filepath)
+
+        data = []
+        for k in sorted(f.keys()):
+            if f[k].attrs["split"] != self.splits[self.split]:
+                continue
```

Comment:
What is this needed for?

Reply:
To skip samples that aren't part of the split